PlaceRouteHierFlow/router/Grid.cpp

```diff
@@ -2530,29 +2530,173 @@ std::vector<int> Grid::Map_from_seg2gridseg_pin(RouterDB::SinkData& sourcelist,
   // SinkData sourceL;
   // find the grid point nearby and insert it to the grid_seg;
 
-  grid_Lx = Lx / (grid_unit_x * grid_scale_func);
+  grid_Lx = Lx;
+  if (drc_info.Metal_info[sourcelist.metalIdx].direct == 0)
```

Comment:
This snippet seems to be repeating multiple times. Wrapping in a function could reduce the number of lines.